src/GenSymIO.chpl

```diff
@@ -644,6 +644,168 @@ module GenSymIO {
+                        hadError = true;
+                        if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
+                    } catch e: PermissionError {
+                        fileErrorMsg = "Permission error %s opening %s".format(e.message(),fname);
+                        gsLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
+                        hadError = true;
+                        if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
+                    } catch e: DatasetNotFoundError {
+                        fileErrorMsg = "Dataset %s not found in file %s".format(dsetname,fname);
+                        gsLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
+                        hadError = true;
+                        if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
+                    } catch e: SegArrayError {
+                        fileErrorMsg = "SegmentedArray error: %s".format(e.message());
+                        gsLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
+                        hadError = true;
+                        if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
+                    } catch e : Error {
+                        fileErrorMsg = "Other error in accessing file %s: %s".format(fname,e.message());
+                        gsLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
+                        hadError = true;
+                        if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
+                    }
+
+                    // This may need to be adjusted for this all-in-one approach
+                    if hadError {
+                      // Keep running total, but we'll only report back the first 10
+                      if fileErrorCount < 10 {
+                        fileErrors.append(fileErrorMsg.replace("\n", " ").replace("\r", " ").replace("\t", " ").strip());
+                      }
+                      fileErrorCount += 1;
+                    }
+                }
+                // This is handled in the readFilesByName() function
+                var subdoms: [filedom] domain(1);
+                var len: int;
+                var nSeg: int;
+                len = + reduce sizes;
+
+                // Only integer is implemented for now, do nothing if the Parquet
+                // file has a different type
+                if ty == ArrowTypes.int64 || ty == ArrowTypes.int32 {
+                  var entryVal = new shared SymEntry(len, int);
+                  readFilesByName(entryVal.a, filenames, sizes, dsetname);
+                  var valName = st.nextName();
+                  st.addEntry(valName, entryVal);
+                  rnames.append((dsetname, "pdarray", valName));
+                }
```

Comment:
Should we return an error tuple if the type is not supported?  I know it's probably already stopped from being sent by the client.

Reply:
Regarding this and the following error-related comments, I was anticipating this is a follow-up PR as Elliot mentioned, but I do agree there are many places errors could be handled better.